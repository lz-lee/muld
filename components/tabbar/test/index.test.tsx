import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import Tabbar from '..';
import TabbarItem from '../../tabbar-item';

const TabbarDemo = (props: any) => {
    return <Tabbar {...props}></Tabbar>;
};

describe('Tabbar', () => {
    afterEach(cleanup);

    it('tabbar value', () => {
        const { asFragment } = render(
            <TabbarDemo active={0}>
                <TabbarItem>Tab</TabbarItem>
                <TabbarItem>Tab</TabbarItem>
            </TabbarDemo>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('click event', () => {
        const onChange = jest.fn();
        const onClick = jest.fn();
        const { container } = render(
            <TabbarDemo active={0} onChange={onChange}>
                <TabbarItem onClick={onClick}>Tab</TabbarItem>
            </TabbarDemo>,
        );
        const tabbarItem = container.querySelector('.mul-tabbar-item');
        fireEvent.click(tabbarItem as Element);
        expect(onChange).toHaveBeenCalledTimes(0);
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    it('name prop', () => {
        const onChange = jest.fn();
        const { container } = render(
            <TabbarDemo active="a" onChange={onChange}>
                <TabbarItem name="a">Tab</TabbarItem>
                <TabbarItem name="b">Tab</TabbarItem>
            </TabbarDemo>,
        );
        const tabbarArr = container.querySelectorAll('.mul-tabbar-item');
        fireEvent.click(tabbarArr[1]);
        expect(onChange).toHaveBeenCalledWith('b');
    });

    it('disable border', () => {
        const { asFragment } = render(
            <TabbarDemo active={0} disable={false}>
                <TabbarItem>Tab</TabbarItem>
            </TabbarDemo>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('placeholder prop', () => {
        const { asFragment } = render(
            <TabbarDemo active={0} fixed={true} placeholder={true}>
                <TabbarItem>Tab</TabbarItem>
            </TabbarDemo>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
