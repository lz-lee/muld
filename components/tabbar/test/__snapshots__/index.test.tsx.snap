// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabbar disable border 1`] = `
<DocumentFragment>
  .c0.mul-tabbar {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: content-box;
  width: 100%;
  height: 3.125rem;
  padding-bottom: constant(safe-area-inset-bottom);
  padding-bottom: env(safe-area-inset-bottom);
  background-color: #fff;
}

.c0.mul-tabbar--fixed {
  position: fixed;
  bottom: 0;
  left: 0;
}

.c0.mul-tabbar--unfit {
  padding-bottom: 0;
}

.c1.mul-tabbar-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #646566;
  font-size: 0.75rem;
  line-height: 1;
  cursor: pointer;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon {
  position: relative;
  margin-bottom: 0.25rem;
  font-size: 1.375rem;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon .mul-icon {
  display: block;
  min-width: 1em;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon img {
  display: block;
  height: 1.25rem;
}

.c1.mul-tabbar-item--active {
  color: #108ee9;
}

.c1.mul-tabbar-item .mul-info {
  margin-top: 0.25rem;
}

<div
    class="c0 mul-hairline--top-bottom mul-tabbar mul-tabbar--fixed"
  >
    <div
      class="c1 mul-tabbar-item mul-tabbar-item--active"
    >
      <div
        class="mul-tabbar-item__icon"
      />
      <div
        class="mul-tabbar-item__text"
      >
        Tab
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabbar placeholder prop 1`] = `
<DocumentFragment>
  .c0.mul-tabbar {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: content-box;
  width: 100%;
  height: 3.125rem;
  padding-bottom: constant(safe-area-inset-bottom);
  padding-bottom: env(safe-area-inset-bottom);
  background-color: #fff;
}

.c0.mul-tabbar--fixed {
  position: fixed;
  bottom: 0;
  left: 0;
}

.c0.mul-tabbar--unfit {
  padding-bottom: 0;
}

.c1.mul-tabbar-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #646566;
  font-size: 0.75rem;
  line-height: 1;
  cursor: pointer;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon {
  position: relative;
  margin-bottom: 0.25rem;
  font-size: 1.375rem;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon .mul-icon {
  display: block;
  min-width: 1em;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon img {
  display: block;
  height: 1.25rem;
}

.c1.mul-tabbar-item--active {
  color: #108ee9;
}

.c1.mul-tabbar-item .mul-info {
  margin-top: 0.25rem;
}

<div
    class="c0 mul-tabbar__placeholder"
    style="height: 0px;"
  >
    <div
      class="c0 mul-hairline--top-bottom mul-tabbar mul-tabbar--fixed"
    >
      <div
        class="c1 mul-tabbar-item mul-tabbar-item--active"
      >
        <div
          class="mul-tabbar-item__icon"
        />
        <div
          class="mul-tabbar-item__text"
        >
          Tab
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabbar tabbar value 1`] = `
<DocumentFragment>
  .c0.mul-tabbar {
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: content-box;
  width: 100%;
  height: 3.125rem;
  padding-bottom: constant(safe-area-inset-bottom);
  padding-bottom: env(safe-area-inset-bottom);
  background-color: #fff;
}

.c0.mul-tabbar--fixed {
  position: fixed;
  bottom: 0;
  left: 0;
}

.c0.mul-tabbar--unfit {
  padding-bottom: 0;
}

.c1.mul-tabbar-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #646566;
  font-size: 0.75rem;
  line-height: 1;
  cursor: pointer;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon {
  position: relative;
  margin-bottom: 0.25rem;
  font-size: 1.375rem;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon .mul-icon {
  display: block;
  min-width: 1em;
}

.c1.mul-tabbar-item .mul-tabbar-item__icon img {
  display: block;
  height: 1.25rem;
}

.c1.mul-tabbar-item--active {
  color: #108ee9;
}

.c1.mul-tabbar-item .mul-info {
  margin-top: 0.25rem;
}

<div
    class="c0 mul-hairline--top-bottom mul-tabbar mul-tabbar--fixed"
  >
    <div
      class="c1 mul-tabbar-item mul-tabbar-item--active"
    >
      <div
        class="mul-tabbar-item__icon"
      />
      <div
        class="mul-tabbar-item__text"
      >
        Tab
      </div>
    </div>
    <div
      class="c1 mul-tabbar-item"
    >
      <div
        class="mul-tabbar-item__icon"
      />
      <div
        class="mul-tabbar-item__text"
      >
        Tab
      </div>
    </div>
  </div>
</DocumentFragment>
`;
